spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -314,6 +320,25 @@ def main(argv: Sequence[str]):
         # Use the result of the current model as additional input of the next model
         fname_prior = fname_seg
 
+        # write JSON sidecar file
+        source_path = os.path.join(path_model, "source.json")
+        if os.path.isfile(source_path):
+            with open(source_path, "r") as fp:
+                source_dict = json.load(fp)
+        sidecar_json = {
+            'GeneratedBy': [
+                {
+                    "Name": f"spinalcordtoolbox: sct_deepseg {' '.join(os.path.basename(arg) for arg in argv)}",
```

Comment:
It seems strange to me that we're applying `basename` even to non-path arguments, but as long as they don't contain `/` or `\` then it should be fine? Especially since the alternative would be to duplicate much of the argument parsing logic.

Reply:
I had the same feeling of weirdness? But, exactly as you pointed out, I figured it would be more straightforward than trying to parse for specific arguments, etc.